### PR TITLE
5-pm-4 Add quarter column to instructor search

### LINKED ISSUE #158: Add Quarter to search by instructor name
# User Story

As a user, I can see the quarter of each class returned by the search by instructor so I can figure out which ones are current.

# Acceptance Criteria
- [x] "Search by Instructor" results table contains a column for quarter.
- [x] Quarters in table correctly correspond to the courses.

# Implementation Todos

Front end:

- [x] Modify table element to add a column for Quarter, populate with quarter information from JSON.

Back end:

Nothing to do

Testing:

- [x] Front end tests pass and there is adequate coverage
- [x] Back end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by boax09**
This pull request closes  #158, "Add Quarter to search by instructor name"

- Column for quarter now is displayed on search by instructor name
- Code was a slight modification from a previous issue, add quarter to search by course name, so only 1 line was added
- Tests already existed from previous epic

### CODE REVIEW

**Review by at527 — APPROVED** [review]
Quarter Column appears on page.
LGTM

**Review by mara-downing — CHANGES_REQUESTED** [review]
Quarter should display as S21, rather than 20212

**Comment by pconrad** [comment]
To address @mara-downing 's concern...  @zminshall has a good suggestion:

Look at this code:

```
const toFormat = (quarter, year) => {
    return year.toString() + (parseInt(quarter)).toString();
}

const fromFormat = (format) => {
    return `${quarters[parseInt(format.charAt(4)) - 1]} ${format.substring(0, 4)}`;
}
```

in https://github.com/ucsb-cs156-w21/proj-ucsb-courses-search/blob/main/javascript/src/main/components/Statistics/QuarterFormSelect.js

Consider factoring those two functions out into a quarterUtilities.js file under the https://github.com/ucsb-cs156-w21/proj-ucsb-courses-search/tree/main/javascript/src/main/utils directory.

You can move the tests into a test file specifically for quarterUtilities.test.js

Then you can use that function in your formatter.

**Comment by boax09** [comment]
> To address @mara-downing 's concern... @zminshall has a good suggestion:
> 
> Look at this code:
> 
> ```
> const toFormat = (quarter, year) => {
>     return year.toString() + (parseInt(quarter)).toString();
> }
> 
> const fromFormat = (format) => {
>     return `${quarters[parseInt(format.charAt(4)) - 1]} ${format.substring(0, 4)}`;
> }
> ```
> 
> in https://github.com/ucsb-cs156-w21/proj-ucsb-courses-search/blob/main/javascript/src/main/components/Statistics/QuarterFormSelect.js
> 
> Consider factoring those two functions out into a quarterUtilities.js file under the https://github.com/ucsb-cs156-w21/proj-ucsb-courses-search/tree/main/javascript/src/main/utils directory.
> 
> You can move the tests into a test file specifically for quarterUtilities.test.js
> 
> Then you can use that function in your formatter.

Formatted column as Professor Conrad mentioned.

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 1]
```diff
@@ -36,8 +37,10 @@ const BasicCourseTable = ( {classes, checks, displayQuarter, allowExport} ) => {
     return (  instructor )
   }
   const renderQuarter = (_cell, row) => {
-    const quarter = (row.section % 100 === 0)? row.course.quarter: "";
-    return (  quarter )
+    const quarter = yyyyqToQyy(row.course.quarter);
+    return (row.section % 100 === 0)? quarter: "";
+    //const quarter = (row.section % 100 === 0)? row.course.quarter: "";
+    //return (  quarter )
```
Comment (at527) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:43]: Delete

File: javascript/src/test/utils/quarterUtilities.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      expect(toFormat("1", "2021")).toBe("20211");
+    });
+
+    test("yyyyqToQyy correctly converts", () => {
+        expect(yyyyqToQyy("20211")).toBe("W21");
+        expect(yyyyqToQyy("20202")).toBe("S20");
+        expect(yyyyqToQyy("20193")).toBe("M19");
+        expect(yyyyqToQyy("20184")).toBe("F18");
+    });
+  });
```
Comment (at527) [inline javascript/src/test/utils/quarterUtilities.test.js:19]: Newline

File: javascript/src/main/utils/quarterUtilities.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+const toFormat = (quarter, year) => {
+    return year.toString() + (parseInt(quarter)).toString();
+}
+
+const fromFormat = (format) => {
+    return `${quarters[parseInt(format.charAt(4)) - 1]} ${format.substring(0, 4)}`;
+}
+
+export { fromFormat, toFormat, yyyyqToQyy};
```
Comment (at527) [inline javascript/src/main/utils/quarterUtilities.js:31]: Newline

**Review by at527 — CHANGES_REQUESTED** [review]
Just a few edits

**Review by at527 — APPROVED** [review]
LGTM

**Review by boax09 — COMMENTED** [review]
Formatted quarter column

**Review by at527 — APPROVED** [review]
Everything passes now.
LGTM

**Review by zminshall — APPROVED** [review]
LGTM

**Pull request merged**